src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -85,6 +108,10 @@ public Double averageSumOdd(List<Integer> numbers) {
      * Задача: вивести все имена кошек в которых хозяева это девушки старше 18 лет</p>
      **/
     public List<String> getCatsNames(List<People> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(owner -> owner.getAge() >= femaleAge
+                        && owner.getSex().equals(People.Sex.WOMEN))
+                .flatMap(cat -> cat.getCatList().stream().map(Cat::getName))
```

Comment:
```suggestion
                .flatMap(cat -> cat.getCatList().stream())
                .map(Cat::getName)
```

Reply:
doesn't make any difference in this case but would be more readable and reduce the complexity of flatMap lambda